motor/core.py

```diff
@@ -323,10 +323,10 @@ def maybe_return_socket(self, sock_info):
             waiter = self.queue.popleft()
             if waiter in self.waiter_timeouts:
                 timeout = self.waiter_timeouts.pop(waiter)
-                self.io_loop.remove_timeout(timeout)
+                self._framework.call_later_cancel(self.io_loop, timeout)
 
-            # TODO: with stack_context.NullContext():
```

Comment:
Can you restore this TODO so I remember to examine it? I'm concerned that if an exception-handler in one Tornado coroutine returns the socket to the pool, another Tornado coroutine that receives the socket will be in the wrong exception-handling context.